controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -266,24 +292,61 @@ func (r *StorageConsumerUpgradeReconciler) removeStorageRequestOwner(ctx context
 	return nil
 }
 
-func (r *StorageConsumerUpgradeReconciler) reconcileStorageConsumer(ctx context.Context, storageCluster *ocsv1.StorageCluster, storageConsumer *ocsv1alpha1.StorageConsumer, consumerConfigMapName string) error {
-	spec := &storageConsumer.Spec
-	spec.ResourceNameMappingConfigMap.Name = consumerConfigMapName
-	spec.StorageClasses = []ocsv1alpha1.StorageClassSpec{
-		{Name: util.GenerateNameForCephBlockPoolStorageClass(storageCluster)},
-		{Name: util.GenerateNameForCephFilesystemStorageClass(storageCluster)},
+func (r *StorageConsumerUpgradeReconciler) reconcileStorageConsumer(
```

Comment:
is there any reason duplicating it here, the minute provider internal consumer is marked as internal storagecluster controller will reconcile it and fills the necessary spec, moreover this is overwritten as well.

let me know if I'm misunderstanding anything.

Reply:
This will be called for both internal consumers and external consumers that were in provider mode

Reply:
ok, that's the missing piece, thanks.